src/Aspire.Dashboard/DashboardWebApplication.cs

```diff
@@ -358,9 +362,70 @@ private static void ConfigureAuthentication(WebApplicationBuilder builder, Dashb
                 };
             });
 
+        if (dashboardStartupConfig.WebAppAuthMode == WebAppAuthMode.OpenIdConnect)
+        {
+            // Configure OpenID Connect (OIDC)
+            authentication.AddScheme<WebAppAuthenticationHandlerOptions, WebAppAuthenticationHandler>(WebAppAuthenticationDefaults.AuthenticationScheme, o =>
+            {
+                o.ForwardDefault = CookieAuthenticationDefaults.AuthenticationScheme;
+                o.ForwardChallenge = OpenIdConnectDefaults.AuthenticationScheme;
+            });
+
+            authentication.AddCookie();
+
+            authentication.AddOpenIdConnect(options =>
+            {
+                // Use authorization code flow so clients don't see access tokens.
+                options.ResponseType = OpenIdConnectResponseType.Code;
+
+                options.SignInScheme = CookieAuthenticationDefaults.AuthenticationScheme;
+
+                // Scopes "openid" and "profile" are added by default, but need to be re-added
+                // in case configuration exists for Authentication:Schemes:OpenIdConnect:Scope.
+                if (!options.Scope.Contains(OpenIdConnectScope.OpenId))
+                {
+                    options.Scope.Add(OpenIdConnectScope.OpenId);
+                }
+
+                if (!options.Scope.Contains("profile"))
+                {
+                    options.Scope.Add("profile");
+                }
+
+                // Redirect to resources upon sign-in.
+                options.CallbackPath = TargetLocationInterceptor.ResourcesPath;
+
+                // Avoid "message.State is null or empty" due to use of CallbackPath above.
+                options.SkipUnrecognizedRequests = true;
+            });
+        }
+
         builder.Services.AddAuthorization(options =>
         {
-            options.AddPolicy(OtlpAuthorization.PolicyName, policy => policy.RequireClaim(OtlpAuthorization.OtlpClaimName));
+            options.AddPolicy(
+                name: OtlpAuthorization.PolicyName,
+                policy: new AuthorizationPolicyBuilder(
+                    OtlpCompositeAuthenticationDefaults.AuthenticationScheme)
+                    .RequireClaim(OtlpAuthorization.OtlpClaimName)
+                    .Build());
+
+            if (dashboardStartupConfig.WebAppAuthMode == WebAppAuthMode.OpenIdConnect)
+            {
+                options.AddPolicy(
+                    name: WebAppAuthorizationDefaults.PolicyName,
+                    policy: new AuthorizationPolicyBuilder(
+                        WebAppAuthenticationDefaults.AuthenticationScheme)
+                        .RequireAuthenticatedUser()
+                        .Build());
+            }
+            else
+            {
+                options.AddPolicy(
+                    name: WebAppAuthorizationDefaults.PolicyName,
+                    policy: new AuthorizationPolicyBuilder()
+                        .RequireAssertion(_ => true)
+                        .Build());
```

Comment:
What about doing what OTLP does and require a claim? The handler registers the required claim if auth mode is Unsecured. Avoids the need for different policies.

Reply:
What would requiring a claim give us? In the OLTP case it seems that any request passing through the `OtlpCompositeAuthenticationHandler` is given that claim during authentication, which is then validated in authorization. Given the authentication scheme is only used by that policy, using a claim seems redundant. I'd vote for removing it from the OLTP code and replacing `OtlpCompositeAuthenticationHandler` with `AddPolicyScheme` instead (similar to 1db68e4984d50fda5f5e80f385eef6bbeb076f61), though maybe I'm missing something.

Reply:
Actually I'm not sure that `OtlpCompositeAuthenticationHandler` can be given the same treatment as it uses multiple authentication schemes.

Reply:
Yes, there are multiple schemes of OTLP.

I like the consistency of auth mode for both endpoints being checked in auth handlers, but what is here is ok if it works. Can always change it in the future if we want.

Reply:
Actually `OtlpCompositeAuthenticationHandler` _can_ be converted to use `AddPolicyScheme` along with `ForwardDefaulSelector`, with a callback to dispatch accordingly. That would clean the code up a bit.